cmd/agent/app/reporter/client_metrics_test.go

```diff
@@ -238,42 +240,46 @@ func TestClientMetricsReporter_Expire(t *testing.T) {
 			SeqNo: nPtr(1),
 		}
 
-		err := tr.r.EmitBatch(batch)
-		assert.NoError(t, err)
-		assert.Len(t, tr.mr.Spans(), 1)
-
-		// here we test that a connected-client gauge is updated to 1 by the auto-scheduled expire loop,
-		// and then reset to 0 once the client entry expires.
-		tests := []struct {
-			expGauge int
-			expLog   string
-		}{
-			{expGauge: 1, expLog: "new client"},
-			{expGauge: 0, expLog: "freeing stats"},
+		getGauge := func() int64 {
+			_, gauges := tr.mb.Snapshot()
+			return gauges["client_stats.connected_clients"]
 		}
-		start := time.Now()
-		for i, test := range tests {
-			t.Run(fmt.Sprintf("iter%d:gauge=%d,log=%s", i, test.expGauge, test.expLog), func(t *testing.T) {
-				// Expire loop runs every 100us, and removes the client after 5ms.
-				// We check for condition in each test for up to 5ms (10*500us).
-				var gaugeValue int64 = -1
-				for i := 0; i < 10; i++ {
-					_, gauges := tr.mb.Snapshot()
-					gaugeValue = gauges["client_stats.connected_clients"]
-					if gaugeValue == int64(test.expGauge) {
-						break
-					}
-					time.Sleep(500 * time.Microsecond)
+
+		t.Run("detect new client", func(t *testing.T) {
+			assert.EqualValues(t, 0, getGauge(), "start with gauge=0")
```

Comment:
Why is this needed? What else could set the gauge to a non zero value earlier?

Reply:
it's a sanity check